lib/spack/spack/analyzers/__init__.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""This package contains code for creating analyzers to extract Application
+Binary Interface (ABI) information, along with simple analyses that just load
+existing metadata.
+"""
+
+from __future__ import absolute_import
+
+import llnl.util.tty as tty
+from .abi import LibabigailAnalyzer
+from .build import (
+    InstallFilesAnalyzer,
+    EnvironmentVariablesAnalyzer,
+    ConfigArgsAnalyzer
+)
+
+import os
+
+__all__ = [
+    'LibabigailAnalyzer',
+    'InstallFilesAnalyzer',
+    'EnvironmentVariablesAnalyzer',
+    'ConfigArgsAnalyzer'
+]
+
+# "all" cannot be an analyzer type
+analyzer_types = {
```

Comment:
look at `llnl.util.lang.list_modules()` and how we use it for most of the places where there are modular directories full o files in Spack.  e.g. `lib/spack/spack/compilers`.  Then look at `spack.util.naming.mod_to_class`. I think you can use the two of these to avoid having this table and imports that need to be updated every time there is a new analyzer.  It'll also make these consistent with a lot of the rest of spack.

Reply:
This would work, I can restructure the analyzers directory to be similar to packages and compilers with one per module file.